src/PowerShellEditorServices/Services/Symbols/Visitors/RegionVisitor.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation.Language;
+using Microsoft.PowerShell.EditorServices.Services.TextDocument;
+
+namespace Microsoft.PowerShell.EditorServices.Services.Symbols
+{
+    internal static class RegionVisitor
+    {
+        internal static void GetRegionsInDocument(ScriptFile file, Func<SymbolReference, AstVisitAction> action)
```

Comment:
I like this a lot, makes reworking it much easier!

Reply:
I assume you meant the separate class, right?
Because the Func-parameter won't make much sense when moving out of ReferenceTable, would it? Or would you prefer a `Action<SymbolReference>` over ex. IEnumerable + LINQ `Concat()`?